### ISSUE TITLE
Support BitBucket server

### ISSUE THREAD

**Issue by rarkins** [mend/renovate-ce-ee#6]
This issue will be used to track the requirements and progress for supporting BitBucket Server (i.e. on-prem/self-hosted). 

Please +1 if you're interested in it and would like to track progress.

**Comment by rarkins** [comment]
Bitbucket Server support has now been added to the main Renovate repository, so Pro support for Bitbucket Server can now follow

**Comment by rarkins** [comment]
FYI the main holdup with Bitbucket Server support is the adding of webhooks.

For GitHub, we use their "app" interface which makes it a piece of cake - the webhook is automatically installed for any repository that installs the app.

For GitLab CE/EE, we recommend a "system" webhook that sends us every webhook, and then we filter them on the Renovate Pro end for repositories and events we are interested in.

For Bitbucket Server, there seems to be no solution that avoids install webhooks individually for every repository, without writing an entire custom app for Bitbucket (in Java) that might be able to do the same.

**Comment by alekseibevzenko** [comment]
Hi, any updates? We're very waiting this feature.

**Comment by ViceIce** [comment]
Webhooks currently are only availabe on repo basis, so here my suggestion:

* autodiscover / configure repositories
* initrepo
* check /update hook for [disabled](https://github.com/renovatebot/renovate/issues/3934) flag changed

I know that this needs repo admin rights.

Other solution:

Provide a simple website from bot where you use browser javascript to fetch available repos and update hooks in with user rights.

**Comment by ViceIce** [comment]
[Web Post Hooks for Bitbucket Server](https://marketplace.atlassian.com/apps/1211539/web-post-hooks-for-bitbucket-server?hosting=server&tab=overview) is able to configure a post receive web hook on project base

**Comment by rarkins** [comment]
@ViceIce do you know for sure if it can be configured per-project?

![image](https://user-images.githubusercontent.com/6311784/61391645-6fd49980-a8bd-11e9-9743-a052d8484c59.png)

Their docs advertise it as per-repository

**Comment by ViceIce** [comment]
I've installed that add-on. Can enable hook on project level hook page.

**Comment by ViceIce** [comment]
![image](https://user-images.githubusercontent.com/1798109/61428990-c7f2b680-a924-11e9-90e9-48adb2ec37ef.png)

**Comment by daekene** [comment]
Apart from web hooks, is there anything else blocking Bitbucket Server support for Pro?

**Comment by rarkins** [comment]
No, that's the main hold-up. I hope to get to it soon, even if it starts with scheduling and job queue only and adds webhooks after.

**Comment by jakauppila** [comment]
I realize this issue is a couple years old but any plans to pick this back up?

**Comment by rarkins** [comment]
Unfortunately not planned

**Comment by uhthomas** [comment]
Hi! What's the current state of Bitbucket Server support? Will it work, but without webhooks or does is it entirely unsupported? Should https://github.com/renovatebot/helm-charts be used instead?

**Comment by rarkins** [comment]
Hi, FYI we are now working on adding Bitbucket Server / Data Center support to Renovate CE+EE. We'll release a public beta of it in an upcoming minor release of the images. If you'd like to get in touch to discuss some of the design decisions we're making, please do!

**Comment by nabeelsaabna** [comment]
https://github.com/mend/renovate-ce-ee/releases/tag/6.8.0
* feat(bitbucket-server): release Beta support for BitBucket-server

https://github.com/mend/renovate-ce-ee/releases/tag/7.4.0
* We are happy to announce Mend Renovate CE/EE for Bitbucket Server has reached General Availability
